src/planner/operator/logical_limit.cpp

```diff
@@ -1,20 +1,82 @@
 #include "planner/operator/logical_limit.h"
 
+#include "binder/expression/literal_expression.h"
+#include "binder/expression/parameter_expression.h"
+#include "common/exception/runtime.h"
+#include "common/type_utils.h"
 #include "planner/operator/factorization/flatten_resolver.h"
 
 namespace kuzu {
 namespace planner {
 
+static uint64_t getLiteralNumber(std::shared_ptr<kuzu::binder::Expression> expr) {
+    uint64_t number = common::INVALID_LIMIT;
+    if (expr == nullptr) {
+        return number;
+    }
+    auto value = common::Value::createDefaultValue(expr->dataType);
+    switch (expr->expressionType) {
```

Comment:
I see multiple place where we have 

```
switch expressionType
case LITERAL
case PARAMETER
return value
```

This is at least the case for query_fts_index, query_hnsw_index and here. If so, we should refactor.

If you decide to take my wild idea, ignore this comment.